### PR TITLE
taming improvements

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

~~riding component now tells you the name of the key you need to hold~~
~~megacarps and bears now dont use shitcode for if they are rideable (i will kill indieana bones for this)~~
(i talked to ryll and they dont want merge conflicts so ill just let them do that in their pr)
tame var is now set to true on the tamed proc, not right before the tamed proc in feeding proc so if an admin atom proc calls tamed the mob is tamed
vatbeast no longer overrides its faction to neutral because it already sets the action to the tamer's

## Why It's Good For The Game

fixes dumb stuf

## Changelog
:cl:
code: improves taming code
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
since you're touching taming code anyway, can you make does (from beps failures) properly tameable? they have a favored food, but feeding them that favored food doesn't actually contribute towards taming them.

**Comment by Fikou** [comment]
> since you're touching taming code anyway, can you make does (from beps failures) properly tameable? they have apples as a favored food for taming, but feeding them apples doesn't actually contribute towards taming them.

maybe the creator just wanted to let you feed apples to the doe and didnt want to add taming featues

**Comment by ATH1909** [comment]
> > since you're touching taming code anyway, can you make does (from beps failures) properly tameable? they have apples as a favored food for taming, but feeding them apples doesn't actually contribute towards taming them.
> 
> maybe the creator just wanted to let you feed apples to the doe and didnt want to add taming featues

they currently don't have (nonstandard) taming features either, though. like, you can't ride them or anything like that.

**Comment by Fikou** [comment]
yeah thats what im saying

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 1]
```diff
@@ -217,7 +217,9 @@
 	. = ..()
 	if(regen_cooldown < world.time)
 		heal_overall_damage(4)
-	if(!rideable && mind)
+	if(!rideable && src.mind)
```
Comment (Jared-Fogle) [inline code/modules/mob/living/simple_animal/hostile/carp.dm]: why did you add src.?
